packages/relay-store/src/Store.ts

```diff
@@ -104,12 +104,20 @@ export class Store extends RelayModernStore {
             return;
         }
         const references = new Set();
+        const legacyIds = [];
         // Mark all records that are traversable from a root
         this._cache.getAllKeys().forEach((id) => {
             const { operation } = this._cache.get(id);
```

Comment:
if we want to introduce backwards compatibility, it would be more correct to recover the old selector so we can avoid deleting data from the cache without users knowing it

this._cache.getAllKeys().forEach((id) => {
            const { operation, selector: oldSelector } = this._cache.get(id);
            const selector = oldSelector || operation.root;
            RelayReferenceMarker.mark((this as any)._recordSource, selector, references, (this as any)._operationLoader);
 });

At the moment there is no test for this case but if you are interested I can guide you in its realization.
what do you think?